web/universal_links.go

```diff
@@ -116,11 +114,18 @@ func webAuthRedirect(c echo.Context) error {
 
 	spaceTrustedDomains := base.Config.TrustedDomains
 	if domains, ok := spaceTrustedDomains[spacePrefix.String()]; ok {
-		for _, domain := range domains {
-			if strings.Contains(parsedRedirect.Host, domain) {
-				return c.Redirect(http.StatusSeeOther, parsedRedirect.String())
-			}
+		if isHostInTheTrustedDomain(parsedRedirect.Host, domains) {
+			return c.Redirect(http.StatusSeeOther, parsedRedirect.String())
 		}
 	}
 	return echo.NewHTTPError(http.StatusBadRequest, "This domain is not allowed to be redirected")
 }
+
+func isHostInTheTrustedDomain(host string, domains []string) bool {
```

Comment:
nit: shouldn't this function be `isHostInTheTrustedDomains()` (with an s at the end)?